source/common/regular-expressions.js

```diff
@@ -14,6 +14,190 @@
+  'getCitationRE': function () {
+    return RegExp(
+      /(\[(?:[^[\]]*@[^[\]]+)\])|(?<=\s|^)(@[\p{L}\d_][\p{L}\d_:.#$%&\-+?<>~/]*)/.source,
+      'gu')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown inline code.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getCodeRE': function () {
+    return RegExp(
+      /`.*?`/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Code Blocks.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getCodeBlockRE': function () {
+    return RegExp(
+      /^\s{0,3}(`{3,}|~{3,})/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches footnotes for project export.
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnExportRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    return RegExp(
+      /\[\^([\w]+?)\]/.source,
+      // Necessary flags + optional multiline flag
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRE': function () {
+    return RegExp(
+      /(\[\^([\da-zA-Z_-]+)\][^:]|\[\^([\da-zA-Z_-]+)\]$)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnReferenceRE': function () {
+    return RegExp(
+      /^\[\^.+\]:\s/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @param   {boolean}  multiline  Whether to match multiline
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFnRefRE': function (multiline = false) {
+    let flag = (multiline) ? 'm' : ''
+    return RegExp(
+      /^\[\^([\da-zA-Z_-]+)\]: (.+)/.source,
+      'g' + flag)
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown footnote references.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getFootnoteRefRE': function () {
+    return RegExp(
+      /\[\^[^\]]+\]/.source,
+      'i')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadRE': function () {
+    return RegExp(
+      /^(#{1,6}) (.*)/.source,
+      'g')
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown Headings.
+   *
+   * @return  {RegExp}              The compiled Regular Expression
+   */
+  'getHeadingRE': function () {
+    return RegExp(
+      /(#+)\s+/.source
+    )
+  },
+
+  /**
+   * Returns a regular expression that matches MarkDown highlighting (???).
```

Comment:
To your question marks: Yeah, this is non-standard syntax that I've added based on a request. It's supported almost nowhere, but the intention is to get certain strips of Markdown displayed textmarker-like (I think Zettlr's themes use yellow background for this)

Reply:
Fantastic. I'll remember that next time I want you to implement something that's not in common mark :P

I'll update the docs!